index.js

```diff
@@ -1039,6 +1039,66 @@
       [Fn(a, $.Boolean), m(a), m(a)],
       Z.filterM);
 
+  //# takeWhile :: (Foldable f, Monoid (f a)) => (a -> Boolean) -> f a -> f a
```

Comment:
btw here `Alternative f` constraint is used to express partiality
https://github.com/guaraqe/total-alternative#readme
https://hackage.haskell.org/package/total-alternative-0.1.0.1/docs/src/Data-List-Alternative.html#head